xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp

```diff
@@ -986,7 +987,9 @@ bool CDVDVideoCodecFFmpeg::GetPictureCommon(VideoPicture* pVideoPicture)
     m_pCodecContext->pix_fmt == AV_PIX_FMT_YUVJ420P)
     pVideoPicture->color_range = 1;
   else
-    pVideoPicture->color_range = 0;
+    pVideoPicture->color_range = m_hints.colorRange == AVCOL_RANGE_JPEG ? 1 : 0;
+
+  CLog::Log(LOGWARNING, "XXX PRI: %d TRC: %d SPC: %d RNG: %d", pVideoPicture->color_primaries, pVideoPicture->color_transfer,  pVideoPicture->color_space, (int)pVideoPicture->color_range);
```

Comment:
Are you sure you want this to be `LOGWARNING`?
Looks more like a candidate for `LOGDEBUG` to me.

Reply:
use C++ style cast